javascripts/discourse/initializers/dark-light-toggle-hamburger.js

```diff
@@ -70,6 +70,7 @@ Have you selected two different themes for your dark/light schemes in user prefe
       lightTheme.media = "none";
 
       Session.currentProp("defaultColorSchemeIsDark", true);
+      Session.currentProp("darkModeAvailable", true);
```

Comment:
What's that for? Is it used (read) in this theme component? 🤔

Reply:
So I found that *if* your system is set to dark mode, and you click `toggle light color scheme` the logo would not change. Adding these `Session.currentProp` methods fixed this. Here is where it is used in core.

https://github.com/discourse/discourse/blob/38bb00edce0764d3e0c4700115622af446c84db5/app/assets/javascripts/discourse/app/widgets/home-logo.js#L37

The only draw back is now, when changing your system from either dark to light, and this component is set to `auto`, the page needs a refresh to load the correct logo. I feel the middle ground I've come to is fine for this component.

I also contemplated turning `off` the auto select.